app/utils/verified-builds.tsx

```diff
@@ -13,29 +16,90 @@ export type OsecRegistryInfo = {
     executable_hash: string;
     last_verified_at: string | null;
     repo_url: string;
-};
-
-export type CheckedOsecRegistryInfo = {
-    explorer_hash: string;
+    verify_command: string;
 };
 
 export function useVerifiedProgramRegistry({
     programId,
+    programAuthority,
     options,
 }: {
     programId: PublicKey;
+    programAuthority: PublicKey | null;
     options?: { suspense: boolean };
 }) {
-    const { data, error, isLoading } = useSWRImmutable(
+    const { url: clusterUrl } = useCluster();
+    const connection = new Connection(clusterUrl);
+
+    const {
+        data: registryData,
+        error: registryError,
+        isLoading: isRegistryLoading,
+    } = useSWRImmutable(
         `${programId.toBase58()}`,
         async (programId: string) => {
-            return fetch(`${OSEC_REGISTRY_URL}/status/${programId}`).then(response => response.json());
+            const response = await fetch(`${OSEC_REGISTRY_URL}/status/${programId}`);
+            return response.json();
         },
         { suspense: options?.suspense }
     );
-    return { data: error ? null : (data as OsecRegistryInfo), isLoading };
+
+    const { program: accountAnchorProgram } = useAnchorProgram(VERIFY_PROGRAM_ID, connection.rpcEndpoint);
+
+    // Fetch the PDA derived from the program upgrade authority
+    // TODO: Add getting verifier pubkey from the security.txt as second option once implemented
+    const {
+        data: pdaData,
+        error: pdaError,
+        isLoading: isPdaLoading,
+    } = useSWRImmutable(
+        programAuthority && accountAnchorProgram ? `pda-${programId.toBase58()}` : null,
+        async () => {
+            const [pda] = PublicKey.findProgramAddressSync(
+                [Buffer.from('otter_verify'), programAuthority!.toBuffer(), programId.toBuffer()],
+                new PublicKey(VERIFY_PROGRAM_ID)
+            );
+            const pdaAccountInfo = await connection.getAccountInfo(pda);
+            if (!pdaAccountInfo || !pdaAccountInfo.data) {
+                console.log('PDA account info not found');
+                return null;
+            }
+            return accountAnchorProgram.coder.accounts.decode('buildParams', pdaAccountInfo.data);
+        },
+        { suspense: options?.suspense }
+    );
+
+    const isLoading = isRegistryLoading || isPdaLoading;
+
+    if (registryError || pdaError) {
+        return { data: null, error: registryError || pdaError, isLoading };
+    }
+
+    // Create command from the args of the verify PDA
+    if (registryData && pdaData && !isLoading) {
+        const verifiedData = registryData as OsecRegistryInfo;
+        verifiedData.verify_command = `solana-verify verify-from-repo -um --program-id ${programId.toBase58()} ${
+            pdaData.gitUrl
```

Comment:
Should we add a marker somewhere telling people it's only possible to verify mainnet programs right now?

Reply:
Sure can do. Any idea on how we should show it? Is there some component for hints or so?